veps/sig-compute/hypervisor-abstraction.md

```diff
@@ -0,0 +1,590 @@
+    }
+   ```
+   Only zero-value fields are set at each layer; `FinalizeVMI` handles derived/status data (CPU topology snapshot, memory status, hotplug sizing, feature dependency resolution). Existing public functions delegate to the resolved provider for backwards compatibility.
+- **Runtime interfaces (`pkg/hypervisor/runtime/`)** – Provides interfaces for tuning a running VM or interacting with the virtualization stack outside of Libvirt. 
+  1.  `DomainTuner` interface for tuning a running virtual machine.
+      ```go
+      type DomainTuner interface {
+          // Adjust memlock limit for QEMU process
+          AdjustResources(vmi *v1.VirtualMachineInstance, additionalOverheadRatio *string) error
+          // Assign housekeeping CPU threads to housekeeping cgroup
+          HandleHousekeeping(vmi *v1.VirtualMachineInstance, domain *api.Domain) error
+      }
+      ```
+
+  2. `HypervisorCapabilityExtractor` interface for querying the hypervisor device for its capabilities.
+      ```go
+      type HypervisorCapabilityExtractor interface {
+        GetHypervFeatures() []string
+      }
+      ```
+
+  3. `VirtLauncherResourceRenderer` for computing resource requirement of the `virt-launcher` pod.
+      ```go
+      type VirtLauncherResourceRenderer interface {
+        GetMemoryOverhead(vmi *v1.VirtualMachineInstance, arch string, additionalOverheadRatio *string) resource.Quantity
+      }
+      ```
+          
+
+- **Converter library (`pkg/virt-launcher/virtwrap/converter/`)** – Adds a new `Converter` interface, which contains the main function to convert VMI to Libvirt domain.
+
+  ```golang
+  type Converter interface {
+    Convert_v1_VirtualMachineInstance_To_api_Domain(vmi *v1.VirtualMachineInstance, domain *api.Domain, c *ConverterContext) (err error)
+  }
+  ```
+
+  The `BaseConverter` struct that implements the above interface would contain common functionality for VMI to domain conversion. Each hypervisor will be able to implement their own converter, e.g., `MshvConverter`, embed the `BaseConverter` and override certain functions of the `BaseConverter` with their own.
+
+- **Node labeller** – Adds parameters `--virt-type` and `--hypervisor-device` to the `cmd/virt-launcher/node-labeller/node-labeller.sh` script, so that it can correctly query Libvirt for node and domain capabilities.
+
+This split preserves the “implement once, reuse everywhere” story without routing everything through a monolithic interface. New hypervisors can land incrementally—start with defaults and webhooks, add converter support, then extend node labelling—while keeping the contract for each area explicit and testable.
+
+### Selection
+
+- `virt-config` loads cluster-wide defaults from an additive `hypervisor` field on the `KubeVirt` CR. The `hypervisor` field is a list of hypervisors that can be supported on the cluster. 
```

Comment:
This relates also to the discussion [here](https://github.com/kubevirt/enhancements/pull/98/files#r2490226840).
I would personally prefer to start from a less intrusive API, one that does not require changing each time we add a different hypervisor or in case we decide to support hybrid cluster hypervisors.